tests/components/media_source/test_local_source.py

```diff
@@ -0,0 +1,66 @@
+"""Test Local Media Source."""
+import pytest
+
+from homeassistant.components import media_source
+from homeassistant.components.media_source import const
+from homeassistant.setup import async_setup_component
+
+
+async def test_async_browse_media(hass):
+    """Test browse media."""
+    assert await async_setup_component(hass, const.DOMAIN, {})
+    await hass.async_block_till_done()
+
+    # Test path not exists
+    with pytest.raises(media_source.BrowseError) as excinfo:
+        await media_source.async_browse_media(
+            hass, f"{const.URI_SCHEME}{const.DOMAIN}/media/test/not/exist"
+        )
+    assert str(excinfo.value) == "Path does not exist."
+
+    # Test browse file
+    with pytest.raises(media_source.BrowseError) as excinfo:
+        await media_source.async_browse_media(
+            hass, f"{const.URI_SCHEME}{const.DOMAIN}/media/test.mp3"
+        )
+    assert str(excinfo.value) == "Path is not a directory."
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6103615)